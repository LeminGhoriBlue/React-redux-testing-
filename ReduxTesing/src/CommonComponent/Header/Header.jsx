import React from 'react'
import { useState } from 'react';
import { NavLink } from "react-router-dom";

function Header(props) {
    const [add, setadd] = useState({
        id: 11,
        name: 3000
    })
    return (
        <div className="addpage">
            <div className='addtext'>{props.title}</div>
            <NavLink
                to={props.pathName}
            >
                <button className='btn-add'>{props.buttonName}</button>
            </NavLink>
        </div>
    )
}

export default Header;