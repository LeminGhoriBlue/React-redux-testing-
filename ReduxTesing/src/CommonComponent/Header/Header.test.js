
import { shallow } from 'enzyme';
import React from 'react';
import Header from './Header';
import Enzyme from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;

let props = {
    title: "Add Data",
    buttonName: 'Add',
    pathName: 'Home'
}

beforeEach(() => {
    wrapper = shallow(<Header {...props} />)
})

describe("<Header/>", () => {
    it("Create a SnapSort", () => {
        expect(wrapper).toMatchSnapshot();
    })

    it("Check Header component", () => {
        const firstElement = wrapper.find('div').first();
        expect(firstElement.prop('className')).toEqual('addpage');
    })

    describe("check header component props", () => {
        it("check Title value", () => {
            const titleValue = wrapper.find('.addtext');
            expect(titleValue.text()).toBe(props.title)
        })

        it("check Button value", () => {
            const buttonName = wrapper.find('.btn-add');
            expect(buttonName.text()).toBe(props.buttonName)
        })

        it("check NavLink Path", () => {
            const NavLinks = wrapper.find('NavLink');
            expect(NavLinks.props().to).toBe(props.pathName)
        })
    })
})