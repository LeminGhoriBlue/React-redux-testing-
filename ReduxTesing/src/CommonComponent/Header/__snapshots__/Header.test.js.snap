// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Create a SnapSort 1`] = `
<div
  className="addpage"
>
  <div
    className="addtext"
  >
    Add Data
  </div>
  <NavLink
    to="Home"
  >
    <button
      className="btn-add"
    >
      Add
    </button>
  </NavLink>
</div>
`;
